subprojects/packagefiles/pcre2/meson.build

```diff
@@ -61,27 +59,24 @@ if not get_option('jit').disabled() and platform_is_unixlike
   cdata.set10('SUPPORT_JIT', true)
 endif
 
-if not get_option('jit_sealloc').disabled()
-  foreach f : ['mkostemp', 'memfd_create', 'secure_getenv']
-    if c_compiler.has_function(
-      f,
-      prefix: '#include <stdlib.h>',
-      args: ['-D_GNU_SOURCE'],
-    )
-      cdata.set10('HAVE_@0@'.format(f.to_upper()), true)
-    endif
-  endforeach
-
-  if cdata.has('HAVE_MKOSTEMP') and cdata.has('HAVE_SECURE_GETENV')
-    if platform_supports_sealloc
-      cdata.set10('SLJIT_PROT_EXECUTABLE_ALLOCATOR', true)
-    elif get_option('jit_sealloc').enabled()
-      error('Your configuration is not supported')
-    else
-      message('jit_sealloc is not supported on this platform')
-    endif
+foreach e : [
+  ['mkostemp', 'stdlib.h'],
+  ['memfd_create', 'sys/mman.h'],
+  ['secure_getenv', 'stdlib.h'],
+]
+  if c_compiler.has_function(
+    e[0],
+    prefix: '#include <@0@>'.format(e[1]),
+    args: ['-D_GNU_SOURCE'],
```

Comment:
I wonder if this should not be simplified to just has_function.

when adding prefix, I've noticed the behavior of has_function changes to be more similar to has_header_symbol. That is, it stops finding the function if adding headers and wrong args.